pkg/util/secret.go

```diff
@@ -54,7 +54,7 @@ func UpsertImmutableSecret(ctx context.Context, client ctrlruntimeclient.Client,
 		return false, err
 	}
 	existing := &coreapi.Secret{}
-	if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: secret.Name}, existing); err != nil {
+	if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: secret.Name, Namespace: secret.Namespace}, existing); err != nil {
```

Comment:
This one surprised me.
How does it work without `Namespace`?

Reply:
Through a namespacing client